### DISCUSSION TITLE
File not found when run in spyder

### DISCUSSION THREAD

**Discussion by VuaeOF** [mkleehammer/pyodbc#1180] [Q&A] [answered]
### Environment

- Python: 3.11.2 (main, Feb 12 2023, 00:48:52) [GCC 12.2.0]
- IPython: 8.5.0
- pyodbc: 4.0.34
- OS: Debian GNU/Linux bookworm/sid
- DB: mysql
- driver: MySQL ODBC Connector 8
- Spyder: 5.4.2

### Issue

I have an ODBC connection set up with all the necessary attributes. It works through isql.
Test code: `import pyodbc; x=pyodbc.connect("DSN=name")`.
This code works in python. It works in ipython run separately.
But it gives an error if run in a spyder ipython window:
```
In[1]: import pyodbc; x=pyodbc.connect("DSN=name")
Traceback (most recent call last):
        Cell In [1], line 1
                import pyodbc; x=pyodbc.connect("DSN=name")
Error: ('01000', "[01000] [unixODBC][Driver Manager]Can't open lib '/usr/lib/x86_64-linux-gnu/odbc/libmyodbc8w.so' : file not found (0) (SQLDriverConnect )")
```

Of course this file exists and is readable.

How to determine what is the difference between running in console and running in spyder?

**Comment by gordthompson** [comment]
Possibly related: https://github.com/mkleehammer/pyodbc/discussions/1157

In that case, the venv was loading its own version of a system library (libcrypto.so) instead of using the system-wide version. The `strace` command was used to discover that.

**Comment by VuaeOF** [comment, marked as answer]
After replacing the driver with mariadb the error disappeared.
The only difference I see is that the path to the mysql driver has /odbc/.
